### PR TITLE
#9721 - Remove `tui-form` classes from demo 

### LINKED ISSUE #9721: 🐞 - Demo should not use any of "tui-" classes other than spacings
### Reproduction url

https://taiga-ui.dev/form

### Description

Some of the core layouting building blocks used in [Forms](https://taiga-ui.dev/form) is missing documentation. These elements (like row, skeleton, table) seem to be useful for a lot of other contexts, but some more documentation how they are intended to use them would be helpful.

### Taiga UI version

4.14.0

### Browser Used

Firefox

### OS Used

Windows

### PR DESCRIPTION
**Pull request by shiv9604**
Partially Fixes #9721 

Description : 
1. Create example 2 in `styles > foundations > form` for migrating tui-form classes to form layout component.
2. Add a deprecation warning.
3. Shift the form component to `layouts > form`.
4. Migrate remaining tui-form classes to form layout component

### CODE REVIEW

File: projects/demo/src/modules/app/demo-routes.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export const DemoRoute = {
     Expand: '/components/expand',
     ElasticContainer: '/components/elastic-container',
     FieldError: '/pipes/field-error',
-    FormLayout: '/components/form',
+    FormLayout: '/layout/form',
```
Comment (waterplea) [inline projects/demo/src/modules/app/demo-routes.ts:62]: Let's remove old Form page completely and rename this to `Form`. Apparently old form page is visited very often and we do not want people to copy paste deprecated example.

File: projects/demo/src/modules/components/multi-select/examples/9/index.html [hunk 2]
```diff
@@ -34,15 +34,14 @@
...
-        <div class="tui-row tui-form__row_multi-fields tui-row_adaptive">
+        <div class="tui-row tui-row_adaptive">
             <label class="tui-col_md-12">
                 Star Wars persons
                 <tui-multi-select
                     [formControl]="testValue"
                     [rows]="1"
                     [tuiTextfieldCleaner]="true"
                     [tuiTextfieldLabelOutside]="true"
-                    [tuiTextfieldSize]="data.textFieldSize"
```
Comment (waterplea) [inline projects/demo/src/modules/components/multi-select/examples/9/index.html:45]: Why did you remove the size?

File: projects/demo/src/modules/markup/form/index.html [hunk 3]
```diff
@@ -94,5 +100,11 @@
             [component]="1 | tuiComponent"
             [content]="1 | tuiExample"
         />
+        <tui-doc-example
```
Comment (waterplea) [inline projects/demo/src/modules/markup/form/index.html:103]: My idea was to recreate 1st example from this page in the layout form component on its page

Comment (shiv9604) [reply projects/demo/src/modules/markup/form/index.html:103]: Okay! We need to move this example 2 to layouts form right?

File: projects/demo/src/modules/app/demo-routes.ts (hunk 1 above)
Comment (shiv9604) [reply projects/demo/src/modules/app/demo-routes.ts:62]: Only layouts form page should exist in demo right?

File: projects/demo/src/modules/components/multi-select/examples/9/index.html (hunk 2 above)
Comment (shiv9604) [reply projects/demo/src/modules/components/multi-select/examples/9/index.html:45]: My Bad! Let me revert it.

**Comment by shiv9604** [comment]
Hi @waterplea & @splincode, I just wanted to keep you posted that we have resolved the PR comments for this PR.

**Review by splincode — APPROVED** [review]

**Review by waterplea — APPROVED** [review]
